lib/NetPacks.h

```diff
@@ -465,7 +465,7 @@ struct SetCommanderProperty : public CPackForClient //120
 	ECommanderProperty which;
 	TExpType amount; //0 for dead, >0 for alive
 	si32 additionalInfo; //for secondary skills choice
-	Bonus accumulatedBonus;
+	std::shared_ptr<Bonus> accumulatedBonus;

```

Comment:
Sorry, missed that line 258 ...
But there are still problems with shared pointers in netpacks - 100% memory leak and valgring will not help there. Each shared_ptr is stored in loadedSharedPointers inside serializer (untill end of game in case of netpacks).